### PR TITLE
Fix UCR expanded column bug when value is `None`

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Addresses http://manage.dimagi.com/default.asp?230228

`ExpandedColumn`s weren't counting `None` results correctly, because the queries that sql agg/sql alchemy were generating always used the `=` operator for comparison, but sql requires the `is` operator for comparing to `NULL`.

@emord 
@czue and @nickpell might be interested too

### CODE REVIEW

File: corehq/apps/userreports/tests/test_columns.py [hunk 1]
```diff
@@ -260,6 +261,35 @@ def test_expansion(self):
...
+        Confirm that expanded columns work when one of the distinct values is None.
+        This is an edge case because postgres uses different operators for comparing
+        columns to null than it does for comparing to non-null values. e.g.
+            "my_column = 4" vs "my_column is NULL"
+        """
+        field_name = 'my_field'
+        submitted_vals = [None, None, 'foo']
+        data_source, column = self._build_report(submitted_vals, field=field_name)
+
+        headers = [column.header for column in data_source.columns]
```
Comment (dimagimon) [inline corehq/apps/userreports/tests/test_columns.py]: F812 list comprehension redefines 'column' from line 273

**Comment by emord** [comment]
Looks good to me, but don't have a lot of context on sqlagg

**Comment by czue** [comment]
nice. 
:+1: on :white_check_mark:

**Comment by emord** [comment]
kicked one of the builds. failed on:

```
======================================================================
FAIL: test_transfer_first_doc_order (corehq.apps.commtrack.tests.test_xml.CommTrackBalanceTransferTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 232, in inner
    return helper(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 205, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 220, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/util/test_utils.py", line 201, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/tests/test_xml.py", line 399, in test_transfer_first_doc_order
    self.check_product_stock(self.sp, product, final, 0)
  File "/mnt/commcare-hq-ro/corehq/apps/commtrack/tests/test_xml.py", line 228, in check_product_stock
    self.assertEqual(expected_soh, latest_trans.stock_on_hand)
AssertionError: Decimal('50.0') != 150
```

**Pull request merged**